test/integration/framework/framework.go

```diff
@@ -177,3 +178,30 @@ func waitForNamespaceDeletion(client kubeclientset.Interface, namespace string)
 	}
 	return nil
 }
+
+func (f *TestFramework) GenerateCustomImagePullSecret(nsName string) error {
+	if TestContext.DockerUserName != "" && TestContext.DockerUserPsd != "" {
+		client := f.GetKubeClient(fmt.Sprintf("%s-create-namespace", f.BaseName))
+		_, err := client.CoreV1().Secrets(nsName).Create(context.TODO(), dockerConfigSecret(CustomImagePullSecretName, nsName, TestContext.DockerUserName, TestContext.DockerUserPsd), metav1.CreateOptions{})
+		if err != nil {
+			Logf("Failed to create secret in the namespace <%s>: %v", nsName, err)
+			return err
+		}
+	}
+	return nil
+}
+
+func dockerConfigSecret(secName, nsName, dockerUserName, dockerUserPassword string) *corev1.Secret {
+	dockerConfig := fmt.Sprintf(`{"auths":{"docker.io":{"username":"%s","password":"%s"}}}`, dockerUserName, dockerUserPassword)
+	return &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secName,
+			Namespace: nsName,
+		},
+		Data: map[string][]byte{
+			corev1.DockerConfigJsonKey: []byte(dockerConfig),
```

Comment:
Have you tested this and its usage by the test code actually. I think this is supposed to be base64 encoded. Does creating a secret via the yaml automatically encodes this?

Reply:
I did test this.  I built a private docker repo on my docker account. The image is the same as the busybox. I specified the image address with my private repo `kevin0204/cus-busybox:test` when the test creates the deployment, it failed to pull the image if I don't specify my docker username/password in the command line, as this is private repo. the credential is required.  On the contrary, if I pass my docker username/password on the command line, the secret gets created, and it can pull the image well. 

Regarding base64 encode,  as we're using secret API to create the secret, we don't need to worry about that.  if create the secret from a yaml file, we have to encode the string.

Reply:
Ack. Thanks for taking time out to test this.